### PR TITLE
Separate activation broadcast receiver from channel handler class

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR fixes **activation handling** code by 
* Fixes some issues with `PushActivationHandlers` singleton, by providing an instance method and converting some static methods into instance methods
* Moves activation broadcast receiver to a separate file, creating methods for registering with callback and unregistering.
* Respects the FlutterEngine lifecycle by registering the receiver when it's attached to engine and deregisters it when it's detached. 

Closes https://github.com/ably/ably-flutter/issues/304

Please note that changes here won't cause a manual application launch, meaning that any message from broadcast receiver will be received only when the Flutter engine is attached. I did this to prevent further bugs raising from inconsistencies related to Flutter engine lifecycle. 
However this is still a temporary solution. A full solution would be to separate third party push plugin so that we can reuse code directly from Firebase library (https://github.com/ably/ably-flutter/issues/292) or completely refactor push notification code on our plugin

Note: Java library will send activation results using broadcasts. That method might have been chosen for native Android apps, but it could be nice if we could have a async alternative that would give results through callback. If so we wouldn't need to deal with broadcast receivers on our plugin.

### CODE REVIEW

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java
Comment (KacperKluka) [inline android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:42]: This can lead to NPEs since you need to first call `instantiate()` before calling this method to assign some value to the `instance` field. However, I see that this is how it has worked before so I wouldn't want to block the PR on this thing :wink:

**Review by KacperKluka — APPROVED** [review]
Overall looks good :+1: One small problem but it was present before so I'll let you decide what to do with it :wink:

File: android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java
Comment (ikbalkaya) [reply android/src/main/java/io/ably/flutter/plugin/push/PushActivationEventHandlers.java:42]: You are right and you are right 😄 This should also be protected by the its use case. Meaning only when flutter engine is attached (meaning methods will be able to communicate over a channel) this will be called. But this will most probably be history for next couple of changes for push notifications

**Review by ikurek — APPROVED** [review]
LGTM 🙂

**Pull request merged**
